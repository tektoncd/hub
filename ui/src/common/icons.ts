export enum Icons {
  Unknown = 'unknown',
  Cat = 'cat',
  Certificate = 'certificate',
  User = 'user',
  Build = 'build',
  Domain = 'domain'
}
