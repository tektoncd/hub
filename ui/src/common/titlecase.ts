/**
 * To convert the first letter of the input word to Upper Case
 */
export const titleCase = (filterName: string) => {
  return filterName.replace(/\w\S*/g, function (word) {
    return word.charAt(0).toUpperCase() + word.substr(1).toLowerCase();
  });
};
