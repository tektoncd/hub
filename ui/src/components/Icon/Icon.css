.hub-icon {
  margin-left: -0.1em;
  vertical-align: -0.15em;
  color: #484848;
}
