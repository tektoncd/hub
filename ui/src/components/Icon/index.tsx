import React from 'react';
import {
  CatIcon,
  CertificateIcon,
  BuildIcon,
  DomainIcon,
  UserIcon,
  IconSize,
  StarIcon,
  GithubIcon
} from '@patternfly/react-icons';
import { Icons } from '../../common/icons';
import './Icon.css';

interface Props {
  id: Icons;
  size: IconSize | keyof typeof IconSize;
  label: string;
}

const Icon: React.FC<Props> = (props: Props) => {
  const { id, size, label } = props;
  switch (id) {
    case Icons.Unknown:
      return <div></div>;
    case Icons.Cat:
      return <CatIcon size={size} className="hub-icon" label={label} />;
    case Icons.Certificate:
      return <CertificateIcon size={size} className="hub-icon" label={label} />;
    case Icons.User:
      return <UserIcon size={size} className="hub-icon" label={label} />;
    case Icons.Build:
      return <BuildIcon size={size} className="hub-icon" label={label} />;
    case Icons.Domain:
      return <DomainIcon size={size} className="hub-icon" label={label} />;
    case Icons.Star:
      return <StarIcon size={size} className="hub-icon" label={label} />;
    case Icons.Github:
      return <GithubIcon size={size} className="hub-icon" label={label} />;
  }
};

export default Icon;
