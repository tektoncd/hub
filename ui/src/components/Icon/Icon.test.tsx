import React from 'react';
import { shallow } from 'enzyme';
import Icon from '.';
import { Icons } from '../../common/icons';

describe('Icon Component', () => {
  it('should render icon for Task Kind', () => {
    const component = shallow(<Icon id={Icons.Build} size="sm" label="Task" />);
    expect(component.debug()).toMatchSnapshot();
    expect(component.find('BuildIcon[label="Task"]').length).toEqual(1);
  });
  it('should render icon for Pipeline Kind', () => {
    const component = shallow(<Icon id={Icons.Domain} size="md" label="Pipeline" />);
    expect(component.debug()).toMatchSnapshot();
    expect(component.find('DomainIcon[label="Pipeline"]').length).toEqual(1);
  });
  it('should render icon for Official Catalog', () => {
    const component = shallow(<Icon id={Icons.Cat} size="lg" label="Official" />);
    expect(component.debug()).toMatchSnapshot();
    expect(component.find('CatIcon[label="Official"]').length).toEqual(1);
  });
  it('should render icon for Verified Catalog', () => {
    const component = shallow(<Icon id={Icons.Certificate} size="xl" label="Verified" />);
    expect(component.debug()).toMatchSnapshot();
    expect(component.find('CertificateIcon[label="Verified"]').length).toEqual(1);
  });
  it('should render icon for Community Catalog', () => {
    const component = shallow(<Icon id={Icons.User} size="sm" label="Community" />);
    expect(component.debug()).toMatchSnapshot();
    expect(component.find('UserIcon[label="Community"]').length).toEqual(1);
  });
  it('should render icon for Category Filter', () => {
    const component = shallow(<Icon id={Icons.Unknown} size="sm" label="CLI" />);
    expect(component.debug()).toMatchSnapshot();
  });
});
