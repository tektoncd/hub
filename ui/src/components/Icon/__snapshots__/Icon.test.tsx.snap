// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should render icon for Category Filter 1`] = `"<div />"`;

exports[`Icon Component should render icon for Community Catalog 1`] = `"<UserIcon size=\\"sm\\" className=\\"hub-icon\\" label=\\"Community\\" color=\\"currentColor\\" noVerticalAlign={false} />"`;

exports[`Icon Component should render icon for Official Catalog 1`] = `"<CatIcon size=\\"lg\\" className=\\"hub-icon\\" label=\\"Official\\" color=\\"currentColor\\" noVerticalAlign={false} />"`;

exports[`Icon Component should render icon for Pipeline Kind 1`] = `"<DomainIcon size=\\"md\\" className=\\"hub-icon\\" label=\\"Pipeline\\" color=\\"currentColor\\" noVerticalAlign={false} />"`;

exports[`Icon Component should render icon for Task Kind 1`] = `"<BuildIcon size=\\"sm\\" className=\\"hub-icon\\" label=\\"Task\\" color=\\"currentColor\\" noVerticalAlign={false} />"`;

exports[`Icon Component should render icon for Verified Catalog 1`] = `"<CertificateIcon size=\\"xl\\" className=\\"hub-icon\\" label=\\"Verified\\" color=\\"currentColor\\" noVerticalAlign={false} />"`;
